File: VerzakovActivityLaba7/app/src/main/java/com/example/activitylifecycle/MainActivity.java
package com.example.activitylifecycle;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.util.Log;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView);

        if(savedInstanceState != null) {
            textView.setText(savedInstanceState.getString("TextToBundle"));
        }
        Log.d("Lifecycle method: ", "OnCreate()");
        textView.append("OnCreate()" + "\n");
    }

    @Override
    protected void onStart() {
        super.onStart();

        Log.d("Lifecycle method: ", "onStart()");
        textView.append("onStart()" + "\n");
    }

    @Override
    protected void onResume() {
        super.onResume();

        Log.d("Lifecycle method: ", "onResume()");
        textView.append("onResume()" + "\n");
    }

    @Override
    protected void onPause() {
        super.onPause();

        Log.d("Lifecycle method: ", "onPause()");
        textView.append("onPause()" + "\n");
    }

    @Override
    protected void onStop() {
        super.onStop();

        Log.d("Lifecycle method: ", "onStop()");
        textView.append("onStop()" + "\n");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        Log.d("Lifecycle method: ", "onDestroy()");
        textView.append("onDestroy()" + "\n");
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);

        Log.d("onSaveInstanceState: ", "onSaveInstanceState()");
        textView.append("onSaveInstanceState()" + "\n");

        outState.putString("TextToBundle", textView.getText().toString());
    }
}

File: VerzakovLaba3/app/src/main/java/com/example/verzakovlab3/MainActivity.kt
package com.example.verzakovlab3

import android.content.SharedPreferences
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.Spinner
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private lateinit var textView10: TextView
    private lateinit var textView5: TextView
    private lateinit var instrumentSpinner: Spinner
    private lateinit var imageView5: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        textView10 = findViewById(R.id.textView10)
        textView5 = findViewById(R.id.textView5)
        instrumentSpinner = findViewById(R.id.instrumentSpinner)
        imageView5 = findViewById(R.id.imageView5)

        val plusButton: Button = findViewById(R.id.plusButton)
        val minusButton: Button = findViewById(R.id.minusButton)
        val button: Button = findViewById(R.id.button)
        val editTextChelikName: EditText = findViewById(R.id.chelik_name)

        button.setOnClickListener {
            // Открываем активность OrderActivity
            val chelikName = editTextChelikName.text.toString()
            val sharedPreferences = getSharedPreferences("MyPrefs", MODE_PRIVATE)
            val editor = sharedPreferences.edit()
            editor.putString("chelikName", chelikName)
            editor.apply()

            val intent = Intent(this, OrderActivity::class.java)
            startActivity(intent)
        }

        val instruments = resources.getStringArray(R.array.instruments)
        val adapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, instruments)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        instrumentSpinner.adapter = adapter

        instrumentSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
                // Обнуляем значения textView5 и textView10 при смене инструмента
                textView5.text = "0"
                textView10.text = "0"

                val selectedInstrument = instruments[position]
                when (selectedInstrument) {
                    "guitar" -> {
                        plusAmount = 3
                        minusAmount = 3
                        imageView5.setImageResource(R.drawable.guitar_image)
                    }
                    "drum" -> {
                        plusAmount = 2
                        minusAmount = 2
                        imageView5.setImageResource(R.drawable.drum_image)
                    }
                    "keyboard" -> {
                        plusAmount = 1
                        minusAmount = 1
                        imageView5.setImageResource(R.drawable.keyboard_image)
                    }
                }

                // Сохраняем выбранный инструмент в SharedPreferences
                val sharedPreferences = getSharedPreferences("MyPrefs", MODE_PRIVATE)
                val editor = sharedPreferences.edit()
                editor.putString("selectedInstrument", selectedInstrument)
                editor.apply()
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
                // Ничего не делаем при отсутствии выбора
            }
        }

        plusButton.setOnClickListener {
            val currentValue10 = textView10.text.toString().toInt()
            textView10.text = (currentValue10 + 1).toString()

            val currentValue5 = textView5.text.toString().toInt()
            textView5.text = (currentValue5 + plusAmount).toString()

            // Сохраняем обновленные значения textView10 и textView5 в SharedPreferences
            val sharedPreferences = getSharedPreferences("MyPrefs", MODE_PRIVATE)
            val editor = sharedPreferences.edit()
            editor.putInt("textView10Value", currentValue10 + 1)
            editor.putInt("textView5Value", currentValue5 + plusAmount)
            editor.apply()
        }

        minusButton.setOnClickListener {
            val currentValue10 = textView10.text.toString().toInt()
            if (currentValue10 > 0) {
                textView10.text = (currentValue10 - 1).toString()
            }

            val currentValue5 = textView5.text.toString().toInt()
            if (currentValue5 >= minusAmount) {
                textView5.text = (currentValue5 - minusAmount).toString()
            }
            val sharedPreferences = getSharedPreferences("MyPrefs", MODE_PRIVATE)
            val editor = sharedPreferences.edit()
            editor.putInt("textView10Value", currentValue10 - 1)
            editor.putInt("textView5Value", currentValue5 - minusAmount)
            editor.apply()
        }
    }

    private var plusAmount = 1
    private var minusAmount = 1
}